File: Nodejs/osc.js
const osc = require("osc");

let udpPort = null;

const init = ({
  addr,
  port,
  onReady,
  onError
}) => {
  
  udpPort = new osc.UDPPort({
    localAddress: "localhost",
    localPort: port+1,
    remoteAddress: addr,
    remotePort: port,
    broadcast: true
  });
  
  udpPort.on("ready", onReady);
  
  udpPort.on("error", onError);
  
  udpPort.open();
};

const sendFloat = ({
  path,
  data
}) => {
  if (udpPort == null) {
    throw new Error("OSC client not initialized!");
  }

  logger.debug(`Sending to path "${path}": ${data}`);

  udpPort.send({
    address: path,
    args: [
      {
        type: "f",
        value: data
      }
    ]
  });
}

module.exports = {
  init,
  sendFloat
}

File: Golang/osc.go
package main

import (
  "github.com/hypebeast/go-osc/osc"
  "github.com/rs/zerolog/log"
)

var oscClient *osc.Client

func initOscClient(addr string, port int) {
  oscClient = osc.NewClient(addr, port)
  log.Info().Msgf("OSC Client ready to toss data to \"%s:%d\"", addr, port)
}

func sendFloat32(path string, data float32) {
  if oscClient == nil {
    log.Panic().Msg("OSC Client not initialized!")
  }

  log.Debug().Msgf("Sending to path %q: %f", path, data)
  msg := osc.NewMessage(path, data)
  oscClient.Send(msg)
}
